ee/codegen/src/project.ts

```diff
@@ -349,11 +349,12 @@ ${errors.slice(0, 3).map((err) => {
 
           nodesToGenerate.push(nodeData);
 
-          const nodeContext = await createNodeContext({
+          const nodeContext = createNodeContext({
             workflowContext: this.workflowContext,
             nodeData,
           });
           this.workflowContext.addNodeContext(nodeContext);
+          await nodeContext.buildProperties();
```

Comment:
Because `createNodeContext` was async, every node would enter that method first before the first node would enter `this.workflowContext.addNodeContext`, which is where the workflow context node-classname reuse logic resided.

I could see us adding the bottom two lines as part of `createNodeContext`, which would limit the blast radius in the test changes and actually allow us to clean up all of the `this.workflowContext.addNodeContext(nodeContext);` lines. but the key is moving the async work from before `addNodeContext` to after `addNodeContext`